managed/models/settings.go

```diff
@@ -52,7 +52,8 @@ type Settings struct {
 	PMMPublicAddress string `json:"pmm_public_address"`
 
 	Updates struct {
-		Disabled bool `json:"disabled"`
+		Disabled       bool `json:"disabled"`
+		LegacyDisabled bool `json:"legacy_disabled"`
```

Comment:
Shouldn't we migrate existing data to match this?

Reply:
The zero value would be `false` which is correct.
Is there any settings specific migration we shall account for?

Reply:
The `disabled` flag is meant for disabling updates altogether while `legacy_disabled` is for disabling `pmm-update` only.
If `disabled: true`, then it's irrelevant how `legacy_disabled` is configured. It will be disabled.

I added comments for more clarity.